### ISSUE TITLE
File history issue in a repository with an enabled sparse checkout

### ISSUE THREAD

**Issue by montoner0** [gitextensions/gitextensions#6892]
## Current behaviour

File history doesn't load (infinite spinner) for files that weren't checked out due to enabled sparse checkout.

## Expected behaviour

File history loaded successfully for such files.

## Steps to reproduce

Create a repository with enabled sparse checkout. E. g.:
```
git init
git remote add origin https://github.com/gitextensions/gitextensions.git
git config core.sparsecheckout true
echo "Setup/*" >> .git/info/sparse-checkout
git pull origin master
```
Then open it with Git Extensions, select any commit, click `File tree` and try to view the file history of any file outside of `Setup` folder.

## Screenshots

Nothing special, actually, `File history` window with a spinner.
![image](https://user-images.githubusercontent.com/13542804/60710699-f7182980-9f1b-11e9-817c-81ed9862b8cc.png)

## Did this work in previous version of GitExtensions

Not tested.

## Environment

- Git Extensions 3.1.1.6049
- Build 2f872105b4eb42421d9fdafd68e90a9e27d20d96
- Git 2.21.0.windows.1
- Microsoft Windows NT 6.1.7601 Service Pack 1
- .NET Framework 4.7.3416.0
- DPI 96dpi (no scaling)

## Diagnostics

Command log doesn't show anything remarkable:
![image](https://user-images.githubusercontent.com/13542804/60710521-90930b80-9f1b-11e9-9155-a11e1067a926.png)

**Comment by pmiossec** [comment]
That is another side effect of #6458.
The current code is allowing to view file history or blame only files that are in the working directory. See https://github.com/gitextensions/gitextensions/issues/6458#issuecomment-481644007

The problem is in `FormFileHistoryController.TryGetExactPath() `

**Comment by gerhardol** [comment]
Closed by #6909
